### ISSUE TITLE
Implement more group_by and window functions

### ISSUE THREAD

**Issue by dreadatour** [iterative/datachain#523]
This is a follow-up issue for the https://github.com/iterative/datachain/issues/228 and https://github.com/iterative/datachain/issues/227.

1. We still have implement the following group_by functions:

- [ ] first
- [ ] last
- [ ] std
- [ ] var
- [ ] min - for not int/float should return None is None in a list, otherwise - any value(like `max([None, None, "Cat", "Dog", None]) --> None`)
- [ ] max - for not int/float should return any value but None (like `max([None, None, "Cat", "Dog", None]) --> "Cat"`)

2. We also need to check and implement missing window-functions, see [this comment](https://github.com/iterative/datachain/pull/515#discussion_r1806870655).

- the list of functions that need to be implemented will be updated

3. We now have SQL functions implemented in two places:

- new `func` module: https://github.com/iterative/datachain/tree/main/src/datachain/lib/func with support of types and signal columns
- old `functions` module: https://github.com/iterative/datachain/tree/main/src/datachain/sql/functions with raw sqlalchemy functions

We need to update the code to use new `func` module only.

**Comment by dmpetrov** [comment]
> We need to update the code to use new `func` module only.

Yes, it should be imported from top level `from datachain.func import min, concat` 

PS: it looks like there are 3 issues in this one :)

**Comment by dreadatour** [comment]
> PS: it looks like there are 3 issues in this one :)

🤔 Indeed. Let me split it.
